swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -144,40 +136,65 @@ class Adapter {
   }
 
   /// Currently disabled resources
-  private var internetResourceEnabled: Bool
+  private var internetResourceEnabled: Bool = false
 
   /// Cache of internet resource
   private var internetResource: Resource?
 
   /// Keep track of resources
   private var resourceListJSON: String?
 
-  /// Starting parameters
-  private let apiURL: String
-  private let token: Token
-  private let id: String
-  private let logFilter: String
-  private let connlibLogFolderPath: String
+  /// Trigger TUN device setup (called after network interface is configured)
+  /// This should be called from onSetInterfaceConfig after applying network settings
+  ///
+  /// Note: Retry logic is handled in Rust's set_tun_from_search() (5 attempts with 100ms delay)
+  func setupTunDevice() {
+    Task {
+      do {
+        try await sessionManager?.setTunFromSearch()
+        Log.log("TUN device set successfully")
+      } catch {
+        Log.error(error)
+      }
+    }
+  }
 
-  init(
```

Comment:
Nit: Can we keep the original `init` / `start` split or does that not work? Would probably reduce the diff a bit further.